Tutorial: Getting started (#590)
tinacms/tinacms.org: content/docs/getting-started/cms-set-up.md

---

> title: Set-up the CMS

Reviewer: ```suggestion
title: Set Up the CMS
```

prev: /docs/getting-started/introduction
next: /docs/getting-started/edit-content
---

> In this step, we will install `tinacms`, create a CMS instance, wrap our app in the `TinaProvider`, and configure the CMS options.

Reviewer 2: Remove comma after "In this step"?
```suggestion
In this step we will install `tinacms`, create a CMS instance, wrap our app in the `TinaProvider`, and configure the CMS options.
```

## Install _tinacms_

> Install the `tinacms` package. This is the core package that you will need anytime you're using Tina. You'll also need to install `styled-components` to render the editing UI properly.

Reviewer: > This is the core package that you will need anytime you're using Tina.

I think this is slightly inaccurate, but I'm not sure if it would be worth explaining the nuance in this beginner-level tutorial

Author: Suggestions?

Reviewer 2: > This is the main package that you will use to create your CMS

```bash
yarn add tinacms styled-components
```

> ## Create a CMS instance, add _TinaProvider_

Reviewer 2: It would be good to mention that at the end of this section the CMS will not be visible

---
title: Set Up the CMS
prev: /docs/getting-started/introduction
next: /docs/getting-started/edit-content
---

In this step we will install `tinacms`, create a CMS instance, wrap our app in the `TinaProvider`, and configure the CMS options.

## Install _tinacms_

Install the `tinacms` package. This is the main package that you will use to create your CMS. You'll also need to install `styled-components` to render the editing UI properly.

```bash
yarn add tinacms styled-components
```

## Create a CMS instance, add _TinaProvider_

The first step to setting up Tina is to create an [instance of `TinaCMS`](/docs/cms). Once created, this [CMS object](/docs/cms#setting-up-the-cms-object) allows us to manage [Plugins](/docs/plugins), [APIs](/docs/apis), [Media Stores](/docs/media), [Events](/docs/events), [Editing UI](/docs/ui) and other key aspects of content management.

Next, wrap your site in the `TinaProvider`. This [provider](https://github.com/tinacms/tinacms/blob/master/packages/tinacms/src/components/TinaProvider.tsx) sets up the editing UI, and provides a [context](https://reactjs.org/docs/context.html) from which we can access the CMS throughout the app.

> Head to `App.js` and follow these steps:

Reviewer: ```suggestion
Open up `src/App.js` and follow these steps:
```

### The Steps

1. Import `TinaProvider` & `TinaCMS`
2. Create an instance of `TinaCMS`
3. Wrap `TinaProvider` around the contents of `App`, pass the `cms`.

**src/App.js**

```diff
import React from 'react';
+import { TinaProvider, TinaCMS } from 'tinacms';
import logo from './Icon.svg';
import './App.css';

function App() {
+ const cms = new TinaCMS();
  return (
+   <TinaProvider cms={cms}>
      <div className="App">
        <PageContent />
      </div>
+   </TinaProvider>
  );
}

export default App;

//...
```

> _Note:_ After this step, the CMS will still not be accessible. We will enable the CMS in a few steps.

## Configure the CMS object

> When instantiating the CMS, you can pass a [CMS options](/docs/cms#cms-configuration) to configure how the CMS object works.

Reviewer: ```suggestion
When instantiating the CMS, you can pass [options](/docs/cms#cms-configuration) to configure how the CMS object works.
```

Update your CMS instance with the `sidebar` option:

**src/App.js**

```js
function App() {
  const cms = new TinaCMS({
    sidebar: true,
  })
  return (
      //...
  )
}
```

### Sidebar

The [CMS options](/docs/cms#cms-configuration) are a **key part of constructing the CMS**. The `sidebar` property controls the _Sidebar UI_ — which renders forms, menus, and other UI that connect with content editing actions. It will not render unless the CMS is enabled.

## Enabling the CMS

The CMS is _disabled by default_. We need to enable the CMS to edit content.

There are various approaches to enable the CMS. One way is to set the `enabled` option in the [CMS configuration](/docs/cms#cms-configuration). Another is to use a `toggle` method to [enable / disable](/docs/cms#disabling--enabling-the-cms) the CMS.

> Let's **wire up the 'Edit This Site' button** to use the `toggle` method. We'll use the `useCMS` hook to get the CMS object from the `TinaProvider`. This hook can be useful throughout your App when needing to [access or update](/docs/cms#accessing-the-cms-object) the CMS. Once we can access the `cms`, we can toggle the enabled state.

Reviewer: ```suggestion
Our demo contains an **"Edit this Site"** button, but you may have noticed that it doesn't do anything right now. Let's wire up this button to enable Tina using the CMS' `toggle` method.

We'll use the `useCMS` hook to get the CMS object from the `TinaProvider`. This hook can be useful throughout your App when needing to [access or update](/docs/cms#accessing-the-cms-object) the CMS. Once we can access the `cms`, we can toggle the enabled state.
```
